cloud-gcp/src/test/java/com/sequenceiq/cloudbreak/cloud/gcp/GCPLoadBalancerMetadataCollectorTest.java

```diff
@@ -0,0 +1,210 @@
+
+    @Mock
+    private Compute compute;
+
+    @Mock
+    private Compute.ForwardingRules forwardingRules;
+
+    @Mock
+    private Compute.ForwardingRules.List forwardingRulesList;
+
+    @Mock
+    private ForwardingRuleList forwardingRuleListResponse;
+
+    @InjectMocks
+    private GcpMetadataCollector underTest;
+
+    @Mock
+    private GcpStackUtil gcpStackUtil;
+
+    private AuthenticatedContext authenticatedContext;
+
+    @BeforeEach
+    public void before() throws IOException {
+        authenticatedContext = GcpMetadataCollectorTest.createAuthenticatedContext();
+        when(gcpStackUtil.getProjectId(any())).thenReturn("gcp-cred");
+        when(gcpComputeFactory.buildCompute(any())).thenReturn(compute);
+        when(compute.forwardingRules()).thenReturn(forwardingRules);
+        when(forwardingRules.list(anyString(), anyString())).thenReturn(forwardingRulesList);
+        when(forwardingRulesList.execute()).thenReturn(forwardingRuleListResponse);
+    }
+
+    @Test
+    public void testCollectSinglePublicLoadBalancer() {
+        List<CloudResource> resources = new ArrayList<>();
+        resources.add(createCloudResource(FORWARDING_RULE_NAME_1, ResourceType.GCP_FORWARDING_RULE));
+        ForwardingRule publicFowardingRule = createPublicFowardingRule();
+        when(forwardingRuleListResponse.getItems()).thenReturn(List.of(publicFowardingRule));
+
+        List<CloudLoadBalancerMetadata> result = underTest.collectLoadBalancer(authenticatedContext, List.of(LoadBalancerType.PUBLIC), resources);
+
+        assertEquals(1, result.size());
+        assertEquals(PUBLIC_IP, result.get(0).getIp());
+        assertEquals(LoadBalancerType.PUBLIC, result.get(0).getType());
+    }
+
+    @Test
+    public void testCollectSinglePublicLoadBalancerWithMultipleForwardingRules() {
+
+        List<CloudResource> resources = new ArrayList<>();
+        resources.add(createCloudResource(FORWARDING_RULE_NAME_1, ResourceType.GCP_FORWARDING_RULE));
+        resources.add(createCloudResource(FORWARDING_RULE_NAME_2, ResourceType.GCP_FORWARDING_RULE));
+        ForwardingRule publicFowardingRule = createPublicFowardingRule();
+        ForwardingRule publicFowardingRule2 = createPublicFowardingRule();
+        publicFowardingRule2.setIPAddress(PUBLIC_IP2);
+        publicFowardingRule2.setName(FORWARDING_RULE_NAME_2);
+        when(forwardingRuleListResponse.getItems()).thenReturn(List.of(publicFowardingRule, publicFowardingRule2));
+
+        List<CloudLoadBalancerMetadata> result = underTest.collectLoadBalancer(authenticatedContext, List.of(LoadBalancerType.PUBLIC), resources);
+
+        assertEquals(2, result.size());
+        String actualIp = result.get(0).getIp();
+        String actualIp2 = result.get(1).getIp();
+        assertTrue((actualIp.equals(PUBLIC_IP2) && actualIp2.equals(PUBLIC_IP)) ||
+                (actualIp.equals(PUBLIC_IP) && actualIp2.equals(PUBLIC_IP2)));
+        assertEquals(PUBLIC_IP, actualIp);
```

Comment:
In the first assert here, you have as an option `(actualIp.equals(PUBLIC_IP2) && actualIp2.equals(PUBLIC_IP))`. But the second assert says `assertEquals(PUBLIC_IP, actualIp)`. So actualIp will never actually equal PUBLIC_IP2, correct?